automate_apex_legends_shader_merge.sh

```diff
@@ -0,0 +1,128 @@
+#!/bin/bash
+
+# Prerequisites for this script:
+# git, git-lfs, dxvk-cache-tool,
+
+# Prerequisites for manual steps: 
+# GitHub account, ssh-key/access-token
+
+manualSteps="Manual steps for merging shaders and updating repo: 
+1. fork this repo: https://github.com/bcook254/apex-legends-cache
+2. git clone the forked repo
+3. cd into cloned repo
+4. git lfs track r5apex.dxvk-cache -> should return \"r5apex.dxvk-cache\" already supported
+5. Copy the shaders from steam shadercache to the local repo: cp ~/.local/share/Steam/steamapps/shadercache/1172470/DXVK_state_cache/r5apex.dxvk-cache ./r5apex-local.dxvk-cache
+6. Use the merge tool to merge your own shadercache with the shadercache files in the repo: dxvk-cache-tool ./r5apex.dxvk-cache ./r5apex-local.dxvk-cache
+7. Extract amount of new entries to the shader from return value of the merge tool
+8. Rename the merge tool output file to the same name of the repo cache files to overwrite the repo cache files: mv ./output.dxvk-cache ./r5apex.dxvk-cache
+9. git add r5apex.dxvk-cache
+10. git commit -m \"\$(date +%s)\" -> might need to enter user.email and user.name
+11. git push origin main -> might need to enter personal access token if ssh is not setup
+12. Go to GitHub, login and go to your forked repository. Then create a new pull request and add a resonable comment. Don't forget to add how many new entries you add with this PR"
+
+# Print help
+
+if [[ "$1" == "-h" || $# -le 2 ]]; then
+    echo "Usage: ";
+    echo "      $0 STEAM_APEX_GAME_DIR LOCAL_REPO_DIR [MERGE_TOOL_DIR]";
+    echo;
+    echo "Best way to find STEAM_APEX_GAME_DIR is to open steam, go to library, right click on Apex Legends and select Manage->Browse local files. Your file browser should open up and you can copy the path to the Apex Legends game directory and supply it to this script."
+    echo;
+    echo "$manualSteps";
+    exit 0;
+fi
```

Comment:
I would like to make this a little more succinct but I'm not sure on our best path forward with that. We certainly need a usage/help message but maybe we can put the manual steps in to a wiki entry?

Reply:
At first the manual steps were only comments inside the script to give me the ability to take a quick look at things I still need to do or that are already done. Later I thought that including the manual steps in the help message could help the user to either quickly self diagonse steps that went wrong while execution and/or to quickly give the user the manual steps in case the execution fails and the reason of failure could not be determined and the user wants to manually do the steps himself. 

But I agree, the manual steps inside the help message might be too long and most likely not neccessary as they are already well described in your repos readme.

Thank you for taking your time to take a closer look at my little script. I appreciate it. I'm looking forward to the refactored/cleaned up version in the next weeks. Let me know if and how I can contribute and help out. Cheers

Reply:
Okay, I decided to remove the manual step from the script it self so a wiki entry can be created later for it.